kernel/include/kernel_internal.h

```diff
@@ -297,6 +297,35 @@ int z_kernel_stats_raw(struct k_obj_core *obj_core, void *stats);
 int z_kernel_stats_query(struct k_obj_core *obj_core, void *stats);
 #endif /* CONFIG_OBJ_CORE_STATS_SYSTEM */
 
+#if defined(CONFIG_THREAD_ABORT_NEED_CLEANUP)
+/**
+ * Perform cleanup at the end of k_thread_abort().
+ *
+ * This performs additional cleanup steps at the end of k_thread_abort()
+ * where these steps require that the thread is no longer running.
+ * If the target thread is not the current running thread, the cleanup
+ * steps will be performed immediately. However, if the target thread is
+ * the current running thread (e.g. k_thread_abort(_current)), it defers
+ * the cleanup steps to later when the work will be finished in another
+ * context.
+ *
+ * @param thread Pointer to thread to be cleaned up.
+ */
+void k_thread_abort_cleanup(struct k_thread *thread);
```

Comment:
Good example here: if your stack creation is a library level thing, you can just provide a top level wrapper or whatever and clean it up on exit.  No need to muck with thread lifecycle stuff in the kernel